### PR TITLE
PGI requires `use iso_c_binding` on module level

### LINKED ISSUE #37: Add `use iso_c_binding` at the top of generated modules
Otherwise the pgi compiler will give a strange error message, something like

>PGF90-S-0099-Illegal use of derived type

### PR DESCRIPTION
**Pull request by havogt**
closes #37

### CODE REVIEW

**Comment by havogt** [comment]
Probably we can now get rid of the `use iso_c_binding` on subroutine level. Fortran expert needed! E.g. here https://github.com/GridTools/cpp_bindgen/blob/f0e926820395e04a62db2222c8848bb755ca266d/tests/regression/array/bindgen_regression_array.f90#L16

File: src/cpp_bindgen/generator.cpp [hunk 1]
```diff
@@ -154,6 +154,7 @@ namespace cpp_bindgen {
     void generate_fortran_interface(std::ostream &strm, std::string const &module_name) {
         strm << "! This file is generated!\n";
         strm << "module " << module_name << "\n";
+        strm << "use iso_c_binding\n"; // pgi wants to have this
```
Comment (mbianco) [inline src/cpp_bindgen/generator.cpp:157]: would this be needed only for PGI or is it good for all the other compilers, too?

**Comment by lxavier** [comment]
@havogt yes if there is already a use iso_c_binding at the module level it is also available in the contained subroutine scope, so you don't need it in the subroutine.

**Comment by lukasm91** [comment]
I don't think we are missing something and I cannot reproduce with PGI. What is the code PGI is complaining about? We have `use iso_c_bindings` in all subroutines / functions where they are actually used, aren't we?

**Comment by havogt** [comment]
Yes, but pgi wants to have it at module scope in some cases. I can give you the example.

File: src/cpp_bindgen/generator.cpp (hunk 1 above)
Comment (havogt) [reply src/cpp_bindgen/generator.cpp:157]: Pgi wants it, all others don’t care.

**Comment by havogt** [comment]
In the interface section the `use iso_c_binding` per subroutine seems to be required for gfortran. Therefore I propose we merge as is (i.e. with the `use iso_c_binding` globally and per subroutine).

**Review by lukasm91 — APPROVED** [review]

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch gridtools

**Pull request merged**